lighthouse-core/audits/manifest-short-name-length.js

```diff
@@ -26,28 +25,37 @@ class ManifestShortNameLength extends Audit {
 
   /**
    * @param {LH.Artifacts} artifacts
-   * @return {Promise<LH.Audit.Product>}
+   * @return {Promise<{failures: Array<string>, manifestValues: LH.Artifacts.ManifestValues, notApplicable?: boolean}>}
    */
-  static audit(artifacts) {
+  static audit_(artifacts) {
+    /** @type {Array<string>} */
+    const failures = [];
+    /** @type {Array<string>} */
+    const warnings = [];
+
     return artifacts.requestManifestValues(artifacts.Manifest).then(manifestValues => {
+      const result = {warnings, failures, manifestValues};
+
+      // If there's no valid manifest, this audit is not applicable
       if (manifestValues.isParseFailure) {
-        return {
-          rawValue: false,
-        };
+        result.notApplicable = true;
+        return result;
+      }
+
+      const shortNameCheck = manifestValues.allChecks.find(i => i.id === 'hasShortName');
+      const shortNameLengthCheck = manifestValues.allChecks.find(i => i.id === 'shortNameLength');
+
+      // If there's no short_name present, this audit is not applicable
+      if (shortNameCheck && !shortNameCheck.passing) {
+        result.notApplicable = true;
+        return result;
       }
 
-      const hasShortName = manifestValues.allChecks.find(i => i.id === 'hasShortName');
-      if (!hasShortName || !hasShortName.passing) {
-        return {
-          rawValue: false,
-          explanation: 'No short_name found in manifest.',
-        };
+      if (shortNameLengthCheck && !shortNameLengthCheck.passing) {
+        failures.push(shortNameLengthCheck.failureText);
       }
 
-      const isShortEnough = manifestValues.allChecks.find(i => i.id === 'shortNameLength');
-      return {
-        rawValue: !!isShortEnough && isShortEnough.passing,
-      };
+      return result;
```

Comment:
there's no warnings and only one possible failure, so why not drop mulitcheckAudit and do like
```js
const manifestValues = await artifacts.requestManifestValues(artifacts.Manifest);
// If there's no valid manifest, this audit is not applicable
if (manifestValues.isParseFailure) {
  return {
    rawValue: true,
    notApplicable: true,
  };
}

const shortNameCheck = manifestValues.allChecks.find(i => i.id === 'hasShortName');
const shortNameLengthCheck = manifestValues.allChecks.find(i => i.id === 'shortNameLength');

// If there's no short_name present, this audit is not applicable
if (shortNameCheck && !shortNameCheck.passing) {
  return {
    rawValue: true,
    notApplicable: true,
  };
}

if (shortNameLengthCheck && !shortNameLengthCheck.passing) {
  return {
    rawValue: false,
    explanation: `Failure: ${shortNameLengthCheck.failureText}.`,
  };
}

return {
  rawValue: true,
};
```